source/xep/httpFileUpload/HttpFileUploadManager.cpp

```diff
@@ -46,13 +47,23 @@ bool HttpFileUploadManager::requestToUploadFileForJid(const QString &file, const
     if (busy_ == false && client_ != nullptr && serverHasFeatureHttpUpload_ == true)
     {
         QFile inputFile(file);
-        QString fileToUpload = createTargetImageName(file);
+        QString fileToUpload = createTargetFileName(file);
 
-        // don't resize image if server can handle it
-        if(inputFile.size() < getMaxFileSize()) 
+        fileType_ = QMimeDatabase().mimeTypeForFile(file).name();
+
+        // Resize image if server cannot handle it or if requested
+        if(fileType_.startsWith("image") && compressImages_ && inputFile.size() > maxImageSize_)
```

Comment:
Just stumbled over this. maxFileSize_ did not get into account any more. The jabber server of your account reports during first connection the maximum file size, it supports for http_upload. This value will be forwarded to maxFileSize_. So, with the new setting, it has to be checked not only for maxImageSize_ of the slider. The image size has to fit also into maxFileuSize_.

Reply:
I will add a test to take into account maxFileSize. Not for compression but to avoid sending a file if it exceeds the maxFileSize.

Reply:
I am not sure about that. What is the purpose of having a slider to set the max compression file size if it does not fit into the limit what the server can handle for an http_upload slot?

Reply:
Agreed. Max value for the slider is max upload size.